### PR TITLE
Move SessionService to Jellyfin.Api

### PR DESCRIPTION
**Pull request by Ullmie02**
Currently not working because I don't fully understand the old underlying infrastructure for the `MediaBrowser.Api` endpoint. Maybe someone with a deeper understanding of the old endpoint/authorization could guide me with my 2 questions.

**Issues**
Part of #2872

### CODE REVIEW

File: Jellyfin.Api/Controllers/SessionController.cs [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+        /// Reports that a session has ended.
+        /// </summary>
+        /// <response code="204">Session end reported to server.</response>
+        /// <returns>A <see cref="NoContentResult"/>.</returns>
+        [HttpPost("/Sessions/Logout")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public ActionResult ReportSessionEnded()
+        {
+            // TODO: how do we get AuthorizationInfo without an IRequest?
+            AuthorizationInfo auth = _authContext.GetAuthorizationInfo(Request);
```
Comment (Ullmie02) [inline Jellyfin.Api/Controllers/SessionController.cs:448]: The new infrastructure has a different `Request` object which I don't think is compatible with `GetAuthorizationInfo`

File: Jellyfin.Api/Helpers/RequestHelpers.cs [hunk 2]
```diff
@@ -25,5 +27,18 @@ internal static string[] Split(string value, char separator, bool removeEmpty)
                 ? value.Split(new[] { separator }, StringSplitOptions.RemoveEmptyEntries)
                 : value.Split(separator);
         }
+
+        internal static SessionInfo GetSession(ISessionContext sessionContext)
+        {
+            // TODO: how do we get a SessionInfo without IRequest?
+            SessionInfo session = sessionContext.GetSession("Request");
```
Comment (Ullmie02) [inline Jellyfin.Api/Helpers/RequestHelpers.cs:34]: The new infrastructure has a different `Request `object which I don't think is compatible with `GetSession`

File: Jellyfin.Api/Controllers/SessionController.cs (hunk 1 above)
Comment (crobibero) [reply Jellyfin.Api/Controllers/SessionController.cs:448]: There will be a new function when the DtoOptions PR is merged that can give you the token

**Comment by Ullmie02** [comment]
Merge after #3363 for working api-docs

**Pull request merged**